jp2_pc/Source/Game/AI/WorldView.cpp

```diff
@@ -367,56 +367,36 @@ void CWorldView::RemoveSomeInfluences()
 							// unimportant enough, go ahead.
 
 	// If we have a bunch of influences, remove one.
-	CInfluenceList::iterator pinf_lowest = inflInfluences.end();
-	CInfluenceList::iterator pinf = inflInfluences.begin();
+	size_t discardable = 0;
 
-	int i_discardable_count = 0;
-	std::list<const CInfluence*> lpinf_discard;
-
-	// Find the lamest influence and remove it.
-	// For now, we only remove one bad apple each cycle.
-	for (; pinf != inflInfluences.end(); pinf++)
+	CInstance* lowestInstance = nullptr;
+	std::list<CInstance*> todelete;
+	
+	for (const auto& entry : inflInfluences)
 	{
-		TReal r_importance = (*pinf).rImportance;
-		
-		if (gaiSystem.sNow - (*pinf).sLastSeen > paniOwner->pbrBrain->sForgetInfluence)
+		if (gaiSystem.sNow - entry.sLastSeen > paniOwner->pbrBrain->sForgetInfluence)
+			entry.setFlags[eifIS_DISCARDABLE] = true;
```

Comment:
Was it at all safe to use an iterator, erase and reassign it from the returned iterator from the erase call, and immediately perform bRemoveInfluence?

Reply:
> A reminder that the main problem with this PR remains that it's trying to push to `master` and not `dev`.

Fixed, switched the target branch